test/format/format.i

```diff
@@ -236,3 +231,38 @@ lock_writeunlock(WT_SESSION *session, RWLOCK *lock)
         testutil_check(pthread_rwlock_unlock(&lock->l.pthread));
     }
 }
+
+/*
+ * trace_worker --
+ *     Write an operations trace message.
+ */
+static inline void
+trace_worker(WT_SESSION *session, const char *fmt, ...)
+{
+    va_list ap;
+
+    va_start(ap, fmt);
+    testutil_check(session->log_vprintf(session, fmt, ap));
+    va_end(ap);
+}
+
+#define tracemsg(fmt, ...)                                                             \
+    do {                                                                               \
+        if (g.trace) {                                                                 \
+            struct timespec __ts;                                                      \
+            __wt_epoch((WT_SESSION_IMPL *)g.trace_session, &__ts);                     \
+            trace_worker(g.trace_session, "[%" PRIuMAX ":%" PRIuMAX "][%s] " fmt,      \
+              (uintmax_t)__ts.tv_sec, (uintmax_t)__ts.tv_nsec / WT_THOUSAND, g.tidbuf, \
+              __VA_ARGS__);                                                            \
+        }                                                                              \
+    } while (0)
+#define traceop(tinfo, fmt, ...)                                                            \
+    do {                                                                                    \
+        if (g.trace) {                                                                      \
+            struct timespec __ts;                                                           \
+            __wt_epoch((WT_SESSION_IMPL *)tinfo->trace, &__ts);                             \
+            trace_worker(tinfo->trace, "[%" PRIuMAX ":%" PRIuMAX "][%s] " fmt,              \
+              (uintmax_t)__ts.tv_sec, (uintmax_t)__ts.tv_nsec / WT_THOUSAND, tinfo->tidbuf, \
+              __VA_ARGS__);                                                                 \
+        }                                                                                   \
```

Comment:
AFAICT the only difference between these two is getting the session. How about refactoring into a macro and have each of these just pass in the session. Then there's only one format string, args and caller to `trace_worker`.

Reply:
They also use different thread-ID buffers.

I went ahead and cleaned up some stuff in 7ebeaa50b, and now I'm no longer using `WT_SESSION.log_vprintf()`. That said, I really think we should offer both, it's standard everywhere as far as I know.